dialoptions.go

```diff
@@ -298,6 +299,16 @@ func WithBlock() DialOption {
 	})
 }
 
+// WithReturnLastError returns a DialOption which makes the client connection
+// return the last connection error that occurred instead of a context.DeadlineExceeded error.
```

Comment:
Please clarify that this returns the connection error string in addition to the context error, not _instead of_ the context error.